### ISSUE TITLE
C# Error function doesn’t work 

### ISSUE THREAD

**Issue by bernatagulloesbrina** [TabularEditor/TabularEditor3#598]
Hello, 
I watched with great interest your presentation at SQLBits 2020, [particularly the part to hack the script adding methods and classes at the end](https://youtu.be/ixRjQ8bbXLw?t=2420). I tried adding a class at the end of the script with the extra } and { and the end but to my surprise these where not expected. I managed to get an execution by just adding the class, but then I'm getting an error when using the Error( )  function, as follows.   It is odd because passing a single string should be enough isn't it? I tried passing some integer and a boolean and now I'm getting a null reference error that I always struggle with. 

Object	Type	Message	Line	Column
	Error	There is no argument given that corresponds to the required formal parameter 'suppressHeader' of 'ScriptHost.Error(string, int, bool)'	85	29

**Comment by otykier** [comment]
The script engine in TE3 allows you to define types and methods without the hack. In fact, the hack won’t work at all, as Roslyn will not compile the script if there are unmatching brackets.

```csharp
class MyClass
{
    public void SayHello()
    {
        Info("Hello");
    }
}

var myClassInstance = new MyClass();
myClassInstance.SayHello();
```

There is a bug with `Error` and a few other global script methods in 3.3.2 (this is also why multiple people experienced issues with macros not working). Fix coming soon!

**Comment by bernatagulloesbrina** [comment]
oh so the behaviour is different in TE2 ? thanks for the quick update and looking forward to the fix! I'll be presenting on c# scripts in september 😁

**Comment by otykier** [comment]
Yes, it’s different. TE3 uses Roslyn, which is faster, more secure and enables the use of new C# 10 language features. This is part of the reason why we upgraded TE3 to use .NET 6.

TE2 uses the old CodeDomProvider for scripting, but this is deprecated these days and only supports language features up to C# 6.

**Comment by bernatagulloesbrina** [comment]
I think I understand the reasons, but at the same time a bit sad that there will be now scripts for TE3 and scripts for TE2. And some (most probably) that will run on both.

**Comment by otykier** [comment]
Yes, it was a priority for us to get TE3 to use Roslyn because of the other features it offers (better IntelliSense, etc.). But the plan is that TE2 will also use Roslyn eventually, thus ensuring that scripts work similar in both tools.

**Comment by bernatagulloesbrina** [comment]
oh, glad to hear that!

**Comment by bernatagulloesbrina** [comment]
I just read that you can also use Rosylin in TE2, right? Will the behaviour be the same then?  
https://github.com/TabularEditor/TabularEditor/releases/tag/2.12.2
however the link with more information seems to be broken

**Comment by bernatagulloesbrina** [comment]
just managed to set it up -- I still need the extra curly brackets but now the behaviour is much closer to TE3

**Comment by otykier** [comment]
Yep, we don't want to ship the Roslyn compiler with TE2, as that will tremendously increase the footprint, and one of the reasons why people love TE2 is that it is so incredibly lightweight. Instead, we will make sure that you can define methods and types just like you can in TE3, but if you want the new C# 10 language features, you'll have to manually configure the Roslyn compiler as described [here](https://docs.tabulareditor.com/te2/Advanced-Scripting.html#compiling-with-roslyn).

**Comment by otykier** [comment]
Issue with `Error` and other global script methods fixed in [3.3.3](https://github.com/TabularEditor/TabularEditor3/releases/tag/3.3.3).
